### DISCUSSION TITLE
"SetOption19 1" does not persist after a device restart?

### DISCUSSION THREAD

**Discussion by akovano** [arendst/Tasmota#18409] [Q&A]
I have a few devices that I do not want to use the Home Assistant Tasmota integration with. I would like these to not be automatically discovered (even if they restart due to power outage) as it causes duplicate entities.
Is there any may to make this setting persist?

**Comment by sfromis** [comment]
As `SetOption19 1` is not supported by the default builds, that flag is not intended to disable Tasmota Discovery, and will be reset upon initialization. Hence, the Tasmota discovery payloads can only be disabled by using a build without this feature. Not using HA, I have no good info there, but maybe you could "tolerate" having the duplicate entries and keep them disabled?

**Reply by akovano** [reply]
Thanks @sfromis but I'm a little confused, hoping you can clarify. I'm using the default 12.4.0 tasmota.bin and setting SetOption19 1 definitely has an effect, the devices immediately get removed from Home Assistant. So if this flag is not intended to disable the Tasmota discovery, it appears to be doing that anyways. Or maybe this will not have any effect in future versions?

**Reply by sfromis** [reply]
What I wrote was not about "having an effect" but about being supported or not. The flag is not intended to disable Tasmota Discovery, but instead to "[use deprecated MQTT discovery](https://tasmota.github.io/docs/Commands/#setoption19)". Hence the flag will be reset during initialization when that feature is no longer included. Not sure if this was a specific intention, but as things are now, that is how it works.

**Reply by arendst** [reply]
I couldn't say it clearer (and that was the intention to keep legacy HASS discovery disabled if tasmota_discovery is in the mix).

**Reply by barbudor** [reply]
Unless I'm reading it wrong as soon as Tasmota Discovery is in the build it automatically set SO19 to 0 here
https://github.com/arendst/Tasmota/blob/c2a6dd81e8d8021bf71259b15c0b974076143f02/tasmota/tasmota_xdrv_driver/xdrv_12_discovery.ino#L252-L257

Which means that command `DiscoveryOff` is likely to have not effect. At next restart or MQTT reconnection, Tasmota Discovery will re-enable itself.

May be we should have:
```C++
    case FUNC_MQTT_INIT:
      if (!Settings->flag.hass_discovery)
            TasDiscoverInit();
      break;
```
